composer/core/state.py

```diff
@@ -108,6 +108,10 @@ class State(Serializable):
     # stopping conditions
     max_epochs: int
 
+    # dataloaders
+    train_dataloader: types.DataLoader
+    eval_dataloader: types.DataLoader
```

Comment:
@anisehsani should be aware of this change since he's working on supporting multiple eval dataloaders

Reply:
Thanks for the ping. This actually simplifies some thing for me!

Reply:
Would be down to discuss this, though my initial thoughts are to keep fit as a zero-argument function, which can allow `__init__` to be used as a kind of static analysis check.